### PR TITLE
Dataset, Repository visibility changes (Do not review)

### PR DESCRIPTION
**Pull request by anandjakhaniya**

### CODE REVIEW

**Comment by conradoverta** [comment]
This PR is massive. Can we break it apart so that we can verify the logic without missing something?

File: backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java [hunk 1]
```diff
@@ -462,7 +465,20 @@ public RepositoryEntity setRepository(
...
+            Optional.empty(), // UAC will populate the owner ID
+            repositoryEntity.isDataset()
+                ? ModelDBServiceResourceTypes.DATASET
+                : ModelDBServiceResourceTypes.REPOSITORY,
+            CollaboratorTypeEnum.CollaboratorType.READ_ONLY,
+            repository.getVisibility());
+        LOGGER.debug("Project role bindings created successfully");
+        Transaction transaction = session.beginTransaction();
+        repositoryEntity.setCreated(true);
+        transaction.commit();
```
Comment (evgeniilezhnin) [inline backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java:481]: what about updating repository visibility and default permissions?

**Review by evgeniilezhnin — COMMENTED** [review]
should there be liquibase changes for created flags?

File: backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java (hunk 1 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java:481]: I hope it will be the same as the project.

Comment (evgeniilezhnin) [reply backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java:481]: I think in repositories you can update visibility, in projects it seems that you can't

**Comment by anandjakhaniya** [comment]
@evgeniilezhnin liquibase change is here https://github.com/VertaAI/modeldb/pull/1772/files#diff-291160cbd97bc2a97c019fed3f6a60ab6044eab86d2105188f12c43df01e4545R358

**Pull request closed without merging**